SmartInsiderTransactionUniverse.cs

```diff
@@ -0,0 +1,126 @@
+
+        /// <summary>
+        /// Market Capitalization in USD
+        /// </summary>
+        public decimal? USDMarketCap { get; set; }
+
+        /// <summary>
+        /// Time passed between the date of the data and the time the data became available to us
+        /// </summary>
+        public TimeSpan Period { get; set; } = TimeSpan.FromDays(1);
+
+        /// <summary>
+        /// Time the data became available
+        /// </summary>
+        public override DateTime EndTime => Time + Period;
+
+        /// <summary>
+        /// Specifies the location of the data and directs LEAN where to load the data from
+        /// </summary>
+        /// <param name="config">Subscription configuration</param>
+        /// <param name="date">Algorithm date</param>
+        /// <param name="isLiveMode">Is live mode</param>
+        /// <returns>Subscription data source object pointing LEAN to the data location</returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            return new SubscriptionDataSource(
+                Path.Combine(
+                    Globals.DataFolder,
+                    "alternative",
+                    "smartinsider",
+                    "transactions",
+                    "universe",
+                    $"{date:yyyyMMdd}.tsv"
+                ),
+                SubscriptionTransportMedium.LocalFile
+            );
+        }
+
+        /// <summary>
+        /// Loads and reads the data to be used in LEAN
+        /// </summary>
+        /// <param name="config">Subscription configuration</param>
+        /// <param name="line">TSV line</param>
+        /// <param name="date">Algorithm date</param>
+        /// <param name="isLiveMode">Is live mode</param>
+        /// <returns>Instance of the object</returns>
+        public override BaseData Reader(SubscriptionDataConfig config, string line, DateTime date, bool isLiveMode)
+        {
+            var csv = line.Split(',');
+            var usdValue = long.Parse(csv[6]);
+            
+            return new SmartInsiderTransactionUniverse
+            {
+                Symbol = new Symbol(SecurityIdentifier.Parse(csv[0]), csv[1]),
+                Time = date - Period,
+                Value = usdValue,
+
+                USDMarketCap = decimal.Parse(csv[2], NumberStyles.Any, CultureInfo.InvariantCulture),
+                MinimumExecutionPrice = decimal.Parse(csv[3], NumberStyles.Any, CultureInfo.InvariantCulture),
+                MaximumExecutionPrice = decimal.Parse(csv[4], NumberStyles.Any, CultureInfo.InvariantCulture),
+                Amount = int.Parse(csv[5]),
+                USDValue = usdValue,
+                BuybackPercentage = decimal.Parse(csv[7], NumberStyles.Any, CultureInfo.InvariantCulture),
+                VolumePercentage = decimal.Parse(csv[8], NumberStyles.Any, CultureInfo.InvariantCulture),
```

Comment:
All those values are declared nullable, but here we are taking for granted that they are not.